sct-commons/src/main/java/org/lfenergy/compas/sct/commons/DoTypeService.java

```diff
@@ -25,4 +32,59 @@ public Optional<TDOType> findDoType(TDataTypeTemplates tDataTypeTemplates, Predi
         return getFilteredDoTypes(tDataTypeTemplates, tdoTypePredicate).findFirst();
     }
 
-}
+    public List<DataAttributeRef> getAllSDOAndDA(TDataTypeTemplates dtt, TDOType tdoType, DataAttributeRef dataRef) {
```

Comment:
Why this method is in `DoTypeService` ? since it does not return any `TDOType`

Reply:
Same answer as  `Why this method is in LnodeTypeService ? since it does not return any TLNodeType`

https://github.com/com-pas/compas-sct/pull/379#discussion_r1528674791

For more details you may look at this [DoTypeAdapter # getDataAttributeRefs](https://github.com/com-pas/compas-sct/blob/develop/sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/dtt/DOTypeAdapter.java#L456)